mesonbuild/build.py

```diff
@@ -1823,8 +1823,7 @@ def process_files(self, files: T.Iterable[T.Union[str, File, 'CustomTarget', 'Cu
             if isinstance(e, CustomTargetIndex):
                 output.depends.add(e.target)
             if isinstance(e, (CustomTarget, CustomTargetIndex)):
-                output.depends.add(e)
```

Comment:
This double add didn't seem to make sense.

Reply:
Something along the lines of
```
stamp_cmd = [find_program('python', required: true), '-c', 'import sys; open(sys.argv[1], "w")', '@OUTPUT0@']
```
might work?